node/src/components/linear_chain_sync.rs

```diff
@@ -176,11 +202,11 @@ impl<I: Clone + PartialEq + 'static> LinearChainSync<I> {
     }
 
     /// Add new block to linear chain.
-    fn add_block(&mut self, block: Block) {
+    fn add_block(&mut self, block_header: BlockHeader) {
         match &mut self.state {
             State::None | State::Done => {}
-            State::SyncingTrustedHash { linear_chain, .. } => linear_chain.push(block),
-            State::SyncingDescendants { linear_chain, .. } => linear_chain.push_back(block),
+            State::SyncingTrustedHash { linear_chain, .. } => linear_chain.push(block_header),
+            State::SyncingDescendants { linear_chain, .. } => linear_chain.push_back(block_header),
```

Comment:
Should we check here that `block_header.parent()` matches `linear_chain.back()`?

Reply:
It should fail the equality check in the other place since hashes won't be the same.